lib/venue.rb

```diff
@@ -1,2 +1,30 @@
 class Venue
+
+  attr_reader :name, :capacity, :patrons
+
+  def initialize(name, capacity, patrons = [])
+    @name = name
+    @capacity = capacity
+    @patrons = patrons
+  end
+
+  def add_patron()
+    @patrons['name']
+  end
+
+  def yell_at_patrons
+    @patrons.upcase
```

Comment:
`@patrons` is an array of Strings, and `upcase` is a method for Strings, so try to iterate over the array so you can get access to each String.